### PR TITLE
Remove/stripe connect functionality

### PR DESCRIPTION
**Pull request by chickenn00dle**
## Description

This removes Stripe onboarding functionality from the plugin altogether.

This is part of the work to move Stripe OAuth code from WC Services over to Stipe. For more information see pb0GrA-Pd-p2

### Testing Instructions

The changes in this PR are dependent on https://github.com/woocommerce/woocommerce-admin/pull/4933. 

Once the above is merged, you can test this by installing WCS and ensuring non-stripe functionality is not broken. For good measure, you can also check that everything works as expected with the various repos from the above P2 checked out locally.

I was unsure about how to test the removal of Stripe code from `client/extensions/woocommerce` directory, so I've removed this in a separate commit just in case this has an impact on something like WordPress.com: https://github.com/Automattic/woocommerce-services/commit/77a209c68cdc0d0ee346117a49c7cab73b379b28

I also noticed Stripe connect code exists in the `wp-calypso` submodule. I haven't opened a PR there to remove this as I wanted to get feedback here first. I'd love some input on this if at all possible.

### Checklist

- [ ] unit tests

One unit test is failing here and on `develop`:

```
 ShippingRate › for all rates › renders the delivery date

    Expected value   <div className="rates-step__shipping-rate-delivery-date">January 1</div>
    Received:
      {}
```

As this test is unrelated to these changes and exists on `develop` I have pushed with `--no-verify` flag.

- [x] `changelog.txt` entry added

### CODE REVIEW

File: client/extensions/woocommerce/app/settings/payments/payment-method-item.js [hunk 1]
```diff
@@ -130,17 +128,7 @@ class PaymentMethodItem extends Component {
 				/>
 			);
 		}
-		if ( method.id === 'stripe' ) {
-			return (
-				<PaymentMethodStripe
-					method={ currentlyEditingMethod }
-					onCancel={ this.onCancel }
-					onEditField={ this.onEditField }
-					onDone={ this.onDoneAndEnable }
```
Comment (frosso) [inline client/extensions/woocommerce/app/settings/payments/payment-method-item.js:139]: Can you please also remove the `onDoneAndEnable` method?

File: client/extensions/woocommerce/app/settings/payments/index.js [hunk 2]
```diff
@@ -51,17 +47,6 @@ class SettingsPayments extends Component {
...
 	};
 
-	componentDidMount = () => {
-		const { site } = this.props;
-
-		// If we are in the middle of the Stripe Connect OAuth flow
-		// go ahead and option the Stripe dialog right away so
-		// we can complete the flow
-		if ( hasOAuthParamsInLocation() || hasOAuthCompleteInLocation() ) {
-			this.props.openPaymentMethodForEdit( site.ID, 'stripe' );
```
Comment (frosso) [inline client/extensions/woocommerce/app/settings/payments/index.js:61]: Can you please also remove the `openPaymentMethodForEdit` prop and the `openPaymentMethodForEdit` function from `mapDispatchToProps`?

**Comment by c-shultz** [comment]
Thanks for the PR! I have this in our review queue now. We'll aim to get this reviewed/merged in the next couple weeks for a patch release on December 10.

File: client/extensions/woocommerce/app/settings/payments/payment-method-item.js (hunk 1 above)
Comment (chickenn00dle) [reply client/extensions/woocommerce/app/settings/payments/payment-method-item.js:139]: Sure thing! Done in https://github.com/Automattic/woocommerce-services/pull/2248/commits/ed7b1b779b6bed719596364d3ff8b45f3172768d

File: client/extensions/woocommerce/app/settings/payments/index.js (hunk 2 above)
Comment (chickenn00dle) [reply client/extensions/woocommerce/app/settings/payments/index.js:61]: Also done in https://github.com/Automattic/woocommerce-services/pull/2248/commits/ed7b1b779b6bed719596364d3ff8b45f3172768d

**Review by frosso — APPROVED** [review]
Tested checkout, labels, tax functionality.
All seems good AFAIK! Just a minor thing on the description of the screenshots.

- DHL live rates
- USPS live rates
- DHL label purchase & refund
- USPS label purchase & refund
- Tax calculations through TaxJar

File: readme.txt [hunk 3]
```diff
@@ -75,9 +71,8 @@ The source code is freely available [in GitHub](https://github.com/Automattic/wo
 2. Setting up custom packages
 3. Selecting your preferred payment method
 4. Enabling automated taxes
-5. Creating a Stripe account from the setup wizard
-6. Checking on the health of WooCommerce Shipping & Tax
-7. Checking and exporting the label purchase reports
+5. Checking on the health of WooCommerce Shipping & Tax
```
[the remark this answers is not in the document]
Comment (chickenn00dle) [reply readme.txt:74]: Sure thing! I've fixed this in https://github.com/Automattic/woocommerce-services/pull/2248/commits/f619fe2edb946d82afb487ad4d4eeea3e1a925f8

**Comment by chickenn00dle** [comment]
Thanks for the review @frosso!

I've taken care of the last comment and this should now be good to go. Should I go ahead and merge or will someone on Heisenberg do this?

**Comment by frosso** [comment]
@chickenn00dle thank you! Feel free to merge, once the merge conflicts are resolved 👍 
Just please ensure to "Squash & Merge" the PR

**Pull request merged**